File: frontend/src/types/order.ts
import type { Courier } from './courier'

export type OrderStatus = 'available' | 'taken' | 'in_transit' | 'delivered' | 'cancelled'

export interface Order {
  id: number
  order_number: string
  address: string
  price: number
  courier_fee: number
  description: string | null
  recipient_phone: string
  admin_phone: string
  status: OrderStatus
  courier_id: number | null
  courier: Courier | null
  latitude: number
  longitude: number
  created_at: string
  updated_at: string
}

export interface OrderCreate {
  order_number: string
  address: string
  price: number
  courier_fee: number
  description?: string
  recipient_phone: string
  admin_phone: string
}

export const statusConfig: Record<OrderStatus, { label: string; color: string }> = {
  available: { label: 'Доступен', color: 'bg-green-100 text-green-800 dark:bg-green-900 dark:text-green-300' },
  taken: { label: 'Взят', color: 'bg-orange-100 text-orange-800 dark:bg-orange-900 dark:text-orange-300' },
  in_transit: { label: 'В пути', color: 'bg-blue-100 text-blue-800 dark:bg-blue-900 dark:text-blue-300' },
  delivered: { label: 'Доставлен', color: 'bg-gray-100 text-gray-800 dark:bg-gray-900 dark:text-gray-300' },
  cancelled: { label: 'Отменён', color: 'bg-red-100 text-red-800 dark:bg-red-900 dark:text-red-300' },
}

File: dostavka_app/lib/models/order.dart
class Order {
  final int id;
  final String orderNumber;
  final String address;
  final double price;
  final double courierFee;
  final String description;
  final String recipientPhone;
  final String adminPhone;
  final String status;
  final double latitude;
  final double longitude;
  final String createdAt;

  const Order({
    required this.id,
    required this.orderNumber,
    required this.address,
    required this.price,
    this.courierFee = 0.0,
    required this.description,
    this.recipientPhone = '',
    this.adminPhone = '+79000000000',
    required this.status,
    required this.latitude,
    required this.longitude,
    required this.createdAt,
  });

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      id: json['id'] as int,
      orderNumber: json['order_number'] as String? ?? '',
      address: json['address'] as String? ?? '',
      price: (json['price'] as num).toDouble(),
      courierFee: (json['courier_fee'] as num?)?.toDouble() ?? 0.0,
      description: json['description'] as String? ?? '',
      recipientPhone: json['recipient_phone'] as String? ?? '',
      adminPhone: json['admin_phone'] as String? ?? '+79000000000',
      status: json['status'] as String? ?? 'available',
      latitude: (json['latitude'] as num).toDouble(),
      longitude: (json['longitude'] as num).toDouble(),
      createdAt: json['created_at'] as String? ?? '',
    );
  }

  String get statusLabel {
    switch (status) {
      case 'available':
        return 'Доступен';
      case 'taken':
        return 'Взят';
      case 'in_transit':
        return 'В пути';
      case 'delivered':
        return 'Доставлен';
      case 'cancelled':
        return 'Отменён';
      default:
        return status;
    }
  }

  bool get isActive => status == 'taken' || status == 'in_transit';
}
